### PR TITLE
testArrayFieldRetrieved

### PR DESCRIPTION
**Pull request by Argentum88**
In order to reproduce the problem described in #416, comment out the following code:
-  `'selectable' => false,`, 
- method `Rebing\GraphQL\Tests\Database\SelectFields\ArrayTests\PostType::resolvePropertiesField()`

### CODE REVIEW

**Comment by mfn** [comment]
I'll take a closer look later, but in the meantime can you please rebase into 1 commit only? Don't think we need the noise.

It should be easy with styleci and involve only one manual step: follow the link where the style violations are reported, download the diff from there and apply it (I usually copy it the to clipboard and paste it from there in PhpStorm).

thanks

**Comment by Argentum88** [comment]
I am sorry for the noise.

File: tests/Support/Models/Post.php [hunk 1]
```diff
@@ -14,6 +14,7 @@
  * @property string $title
  * @property string|null $body
  * @property int|null $user_id
+ * @property string|null $properties
```
Comment (mfn) [inline tests/Support/Models/Post.php]: Due to the `$casts` this is actually supposed to be `array|null`

File: tests/Database/SelectFields/ArrayTests/ArrayTest.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Rebing\GraphQL\Tests\Database\SelectFields\ArrayTests;
+
+use Rebing\GraphQL\Tests\TestCaseDatabase;
+use Rebing\GraphQL\Tests\Support\Models\Post;
+
+class ArrayTest extends TestCaseDatabase
+{
+    public function testArrayFieldRetrieved()
```
Comment (mfn) [inline tests/Database/SelectFields/ArrayTests/ArrayTest.php]: Please add `: void` return type

File: tests/Database/SelectFields/ArrayTests/ArrayQuery.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    public function resolve($root, $args, $ctx, ResolveInfo $info, Closure $getSelectFields)
+    {
+        /** @var SelectFields $selectFields */
+        $selectFields = $getSelectFields();
+
+        return Post
+            ::with($selectFields->getRelations())
+            ->select(['*'])
```
Comment (mfn) [inline tests/Database/SelectFields/ArrayTests/ArrayQuery.php]: Remove

File: tests/Database/SelectFields/ArrayTests/ArrayQuery.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            //->select($selectFields->getSelect())
```
Comment (mfn) [inline tests/Database/SelectFields/ArrayTests/ArrayQuery.php]: Enable

File: tests/Database/SelectFields/ArrayTests/PostType.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    public function fields(): array
+    {
+        return [
+            'id' => [
+                'type' => Type::nonNull(Type::ID()),
+            ],
+            'properties' => [
+                'type' => Type::listOf(GraphQL::type('Property')),
+                'selectable' => false,
```
Comment (mfn) [inline tests/Database/SelectFields/ArrayTests/PostType.php]: Replace this with `'is_relation' => false,` => this is also your fix for https://github.com/rebing/graphql-laravel/issues/416 I believe!

File: tests/Database/SelectFields/ArrayTests/PostType.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            ],
+        ];
+    }
+
+    protected function resolvePropertiesField($root, $args)
```
Comment (mfn) [inline tests/Database/SelectFields/ArrayTests/PostType.php]: Remove

**Comment by mfn** [comment]
Oh, one more suggestion: I think it would be easier for actual collaboration if you used branches and not master for this (technically probably doesn't matter, but that would allow clean working together without the angst of modifying another ones master branch); thanks.

**Comment by Argentum88** [comment]
`is_relation` is a solution, it is even specified in the documentation. Thank you for your help!

**Comment by Argentum88** [comment]
>I think it would be easier for actual collaboration if you used branches and not master for this

But merge is possible only into an existing branch of upstream repo. How separate branch in my fork can be help?

Maybe I don’t know something, because PR is very rare in my practice.

**Comment by mfn** [comment]
> But merge is possible only into an existing branch of upstream repo. How separate branch in my fork can be help?
> 
> Maybe I don’t know something, because PR is very rare in my practice.

On a clone of this repo, which your https://github.com/Argentum88/graphql-laravel is, you just push a branch with any code and **on this** repo you will also have the button to create a PR from that _against this repo master_.

It's described as `rebing:master` <-> `Argentum88:feature-branch` or something and works well :-)

**Comment by Argentum88** [comment]
What is the practical use of this?
This can be useful if you or someone else is working with `Argentum88:another-feature-branch`.

**Comment by mfn** [comment]
Many thanks 👍

**Comment by mfn** [comment]
> What is the practical use of this?

Not sure the question .. that's how it's done on github, git and feature branches 🤷‍♀️

**Pull request merged**